src/components/molecules/CastFrames.tsx

```diff
@@ -0,0 +1,219 @@
+        }
+    }, []);
+
+    function fetchWithTimeout(url: string, options: RequestInit, timeout: number = 8000): Promise<Response> {
+        return Promise.race([
+            customFetch(url, options),
+            new Promise<Response>((_, reject) =>
+                setTimeout(() => reject(new Error('Request timed out')), timeout)
+            )
+        ]);
+    }
+
+    const handleFrameBtnPress = async (btnIndex: number) => {
+        if (!signerValue) {
+            alert('Signer UUID is not available');
+            return;
+        }
+
+        const button = localFrame.buttons.find(btn => btn.index === btnIndex);
+        const postUrl = button?.post_url;
+
+        try {
+            const response = await fetchWithTimeout(`${NEYNAR_API_URL}/v2/farcaster/frame/action?client_id=${client_id}`, {
+                method: "POST",
+                headers: {
+                    "accept": "application/json",
+                    "content-type": "application/json"
+                },
+                body: JSON.stringify({
+                    "cast_hash": hash,
+                    "signer_uuid": signerValue,
+                    "action": {
+                        "button": {
+                            "index": btnIndex
+                        },
+                        "frames_url": localFrame.frames_url,
+                        "post_url": postUrl ? postUrl : localFrame.frames_url
+                    }
+                })
+            }) as Response;
+            if (response.ok) {
+                const json = await response.json() as NeynarFrame;
+                setLocalFrame(json);
+            } else {
+                throw new Error(`HTTP error! status: ${response.status}`);
+            }
+        } catch (error) {
+            console.error('Error during fetch:', error);
+            alert('An error occurred!');
+        }
+    };
+
+    const renderFrameButtons = () => {
+        const buttons = [];
+        const buttonTags = localFrame.buttons;
+
+        if (!buttonTags) {
+            return null;
+        }
+
+        for (let i = 0; i < buttonTags.length; i++) {
+            buttons.push(
+                <CastFrameBtn key={buttonTags[i].index} number={buttonTags[i].index} text={buttonTags[i].title} actionType={buttonTags[i].action_type} target={buttonTags[i].target} handleOnClick={handleFrameBtnPress} />
+            );
+        }
+
+        return <ButtonContainer>{buttons}</ButtonContainer>;
+    };
+
+    const extractDomain = (url: string) => {
+        try {
+            return new URL(url).hostname.replace('www.', '');
+        } catch (error) {
+            return '';
+        }
+    };
+
+    return (
+        <>
+            <FrameContainer>
+                {localFrame !== null && localFrame.frames_url &&
+                <>
+                    <a href={localFrame.frames_url} target="_blank" rel="noopener noreferrer">
```

Comment:
Should this be an a tag or Link tag, I don't remember the pros/cons of both on top of my mind bt please refer the NextJs docs.

I remember there were some caveats. They might have fixed it in latest version not sure.

Reply:
I don't think we have access to Link since it's a part of Next.js(which isn't a dependency), but I don't think we should have any issues with using a native `<a>` tag

Reply:
Oh, correct, my bad. I must’ve been in some Next.js fantasy land, not React reality.

Reply:
Vercel is doing a really get job on encapsulating React.